graphix/sim/base_backend.py

```diff
@@ -1,10 +1,77 @@
 from __future__ import annotations
 
+import typing
+from dataclasses import dataclass
+
 import numpy as np
 
 import graphix.clifford
-import graphix.command
 import graphix.pauli
+import graphix.states
+from graphix.command import CommandKind
+from graphix.ops import Ops
+
+if typing.TYPE_CHECKING:
+    import collections
+
+    from graphix.pauli import Plane
+
+
+@dataclass
+class MeasurementDescription:
+    plane: Plane
+    angle: float
+
+
+class NodeIndex:
```

Comment:
what's wrong with a `list` as `node_index`, as in previous implementation? Perhaps it is specific to BQC (which is fine)? I'd appreciate it if you could a few lines of description to this class.